File: js/euler78.js
function euler78(){
    const sums = new Array(60001).fill(0);
    sums[0] = 1;
    
    for (let i = 1; i < sums.length; i++){
        for (let n = i; n < sums.length; n++){
            sums[n] += sums[n - i];
            sums[n] %= 1_000_000;
        }
        if (sums[i] == 0) return i;
    }
}

console.log(euler78());

File: go/euler78.go
package main
import "fmt"

func main() {
    n := 60000
    sums := make([]int, n + 1)
    sums[0] = 1
    for i := 0; i < len(sums); i++ {
        for j := i; j < len(sums); j++ {
            sums[j] += sums[j - i] % 1000000
        }
        sums[i] %= 1000000
        if sums[i] == 0 {
            fmt.Println(i)
            break
        }
    }
}
